addons/im_livechat/static/src/embed/common/chat_window_patch.js

```diff
@@ -1,18 +1,53 @@
+import { SESSION_STATE } from "./livechat_service";
 import { FeedbackPanel } from "@im_livechat/embed/common/feedback_panel/feedback_panel";
+import { CloseConfirmation } from "@im_livechat/embed/common/close_confirmation";
 
 import { ChatWindow } from "@mail/core/common/chat_window";
 
-import { useState } from "@odoo/owl";
+import { useState, toRaw } from "@odoo/owl";
 
 import { useService } from "@web/core/utils/hooks";
 import { patch } from "@web/core/utils/patch";
 
-Object.assign(ChatWindow.components, { FeedbackPanel });
+Object.assign(ChatWindow.components, { FeedbackPanel, CloseConfirmation });
 
 patch(ChatWindow.prototype, {
     setup() {
         super.setup(...arguments);
         this.livechatService = useService("im_livechat.livechat");
         this.chatbotService = useState(useService("im_livechat.chatbot"));
+        this.livechatState = useState({
+            showCloseConfirmation: false,
+        });
+    },
+
+    async close() {
+        if (this.thread?.channel_type !== "livechat") {
+            return super.close();
+        }
+        const chatWindow = toRaw(this.props.chatWindow);
+        if (chatWindow.folded) {
+            this.props.chatWindow.open({ notifyState: this.thread?.state !== "open" });
+        }
+        this.livechatState.showCloseConfirmation = true;
+        if (this.props.chatWindow.closeFeedback) {
+            await super.close();
+        }
```

Comment:
Close confirmation should happen when the thread is persisted (see `livechat_service.SESSION_STATE`).

When visitor opens livechat without posting any message, there's no real conversation so there's no point in showing the confirmation dialog.